src/gpuarray_buffer.c

```diff
@@ -121,8 +121,10 @@ int gpudata_transfer(gpudata *dst, size_t dstoff, gpudata *src, size_t srcoff,
 
   /* Fallback to host copy */
   tmp = malloc(sz);
-  if (tmp == NULL)
-    return error_sys(ctx->err, "malloc");
+  if (tmp == NULL) {
+    error_sys(src_ctx->err, "malloc");
+    return error_sys(dst_ctx->err, "malloc");
```

Comment:
If they are the same context, I suppose it is fine to set the error twice. The slow down is probably not important.